### PR TITLE
drivers: can: add API for getting supported capabilities

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Add API function for getting the supported capabilities of a CAN controller. This allows for writing portable CAN applications.
This is a preparation step for adding more `CAN_MODE_*` flags.

### CODE REVIEW

File: drivers/can/can_handlers.c [hunk 1]
```diff
@@ -138,6 +138,14 @@ static inline int z_vrfy_can_get_max_filters(const struct device *dev, enum can_
 }
 #include <syscalls/can_get_max_filters_mrsh.c>
 
+static inline int z_vrfy_can_get_capabilities(const struct device *dev, can_mode_t *cap)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_CAN(dev, get_capabilities));
+
```
Comment (str4t0m) [inline drivers/can/can_handlers.c:144]: We should make sure the thread has write access.
`Z_OOPS(Z_SYSCALL_MEMORY_WRITE(cap, sizeof(*cap))); `

**Review by str4t0m — CHANGES_REQUESTED** [review]
Looks good, would only add one check.

File: drivers/can/can_handlers.c (hunk 1 above)
Comment (henrikbrixandersen) [reply drivers/can/can_handlers.c:145]: Nice catch, forgot about that. Added.

**Review by str4t0m — APPROVED** [review]

**Review by aaillet — APPROVED** [review]
Renesas part LGTM

**Pull request merged**